Fix `without_inactive_zero_length_elements` method to only remove elements with exact zero length (#633)
desy-ml/cheetah: CHANGELOG.md

- Fix issue, where the first example in the gradient-based optimisation notebook from the docs would not run because the `k1` of the quadrupoles was set to `0.0`, which causes the gradients to be undefined (see #588, #603) (@jank324, @cr-xu)
- Update the elegant conversion to include missing dipole attributes, converting `hgap` to `gap` and `fint` to `fringe_integral`. (see #624) (@cr-xu)
- Update the `to_openpmd_particlegroup` conversion to use `int` for status and add `detach()` before tensor to numpy conversion. (see #629) (@roussel-ryan)

> - Fixes the issue that negative length elements are incorrectly removed by the `Segment.without_inactive_zero_length_elements` method (see #633) (@cr-xu)

Reviewer: ```suggestion
- Fix an issue where negative length elements are incorrectly removed by the `Segment.without_inactive_zero_length_elements` method (see #633) (@cr-xu)
```

> - Fixes the issue that negative length elements are incorrectly removed by the `Segment.without_inactive_zero_length_elements` method. (see #633) (@cr-xu)

Reviewer: ```suggestion
- Fixes the issue that negative length elements are incorrectly removed by the `Segment.without_inactive_zero_length_elements` method (see #633) (@cr-xu)
```
